staging/src/k8s.io/legacy-cloud-providers/gce/gce_disks.go

```diff
@@ -620,6 +624,68 @@ func (g *Cloud) DisksAreAttached(diskNames []string, nodeName types.NodeName) (m
 	return attached, nil
 }
 
+// BulkDisksAreAttached is a batch function to check if all corresponding disks are attached to the
```

Comment:
there are quite some loop functions  in this BulkDisksAreAttached function, I am wondering whether we can reduce some to improve the performance.

Reply:
While there are a lot of for loops, the run time should still be O(numNodes + numDisks). Some of the loops are a bit tricky - despite being a double for loop it still looks over only as many items as there are disks or nodes. 

Assuming that the runtime for getInstancesByName is constant. What's really going to slow it down is probably making API queries if getInstancesByName's heuristic is not good.